engine/runtime/src/test/scala/org/enso/compiler/test/codegen/AstToIrTest.scala

```diff
@@ -213,6 +213,38 @@ class AstToIrTest extends CompilerTest with Inside {
       val pattern = ir.branches.head.pattern
       pattern shouldBe an[IR.Pattern.Constructor]
     }
+
+    "support literal numeric patterns" in {
+      val ir =
+        """
+          |case foo of
+          |    1 -> 10
+          |    2 -> 20
+          |""".stripMargin.toIrExpression.get.asInstanceOf[IR.Case.Expr]
+
+      ir.branches(0).pattern shouldBe an[IR.Pattern.Literal]
+      ir.branches(1).pattern shouldBe an[IR.Pattern.Literal]
+    }
+
+    "support constructor patterns with nested literals" in {
+      val ir =
+        """
+          |case foo of
+          |    Cons 1 b -> a + b
```

Comment:
Interesting that `Cons 1 b` works!

Reply:
It parses, it doesn't necessairly work :p